### PR TITLE
Data path healing should be disabled

### PR DESCRIPTION
**Pull request by ljkiraly**
Related issue: #446

Signed-off-by: Laszlo Kiraly <laszlo.kiraly@est.tech>

### CODE REVIEW

File: internal/config/config.go [hunk 1]
```diff
@@ -44,6 +44,7 @@ type Config struct {
 
 	CoreDNSConfigPath string `default:"/etc/coredns/Corefile" desc:"Default path to the Corefile source"`
 
+	LivenessCheck         string        `default:"enabled" desc:"Enable/Disable dataplane liveness check"`
```
[the remark this answers is not in the document]
Comment (uablrek) [reply internal/config/config.go:47]: lgtm beside the comment above

**Comment by edwarnicke** [comment]
@uablrek I'm completely fine with an option to disable DataPath healing check... I'm curious though *why* :)  I suspect there's an interesting issue/usecase there :)

**Comment by ljkiraly** [comment]
@edwarnicke The new data path healing in NSC check the connectivity to NSE by ping. In case of nse-remote-vlan the NSE doesn't play any role in data path. The check always fails and the connection is reset by healing (by "Reconnect with reselect") in NSC

We considered the following solutions for this issue with nse-remote-vlan:
* disable the data path healing and/or
* set a connection endpoint address which can be used for data path monitoring.

1/ Healing should be disabled in NSC by environment parameter.
2/ Healing should be disabled in nse-remote-vlan by setting only 'src' address in IP context (no 'dst' address or same 'dst' as 'src')
3/ Healing should be supported by setting one (or more) remote address in nse-remote-vlan as environment parameter and passing by IP context. Adds data path supervision (maybe a bit expensive).

We prefer to monitor this connection by BGP and/or BFD, so the first option was chosen. The first implementation proposal seemed cleaner compared to 2nd.
Also got a comment that would be preferable that the status on the user (BFD) layer could be signaled to the NSC to trigger some kind of remedy of the lower (NSC) layer, if possible - this might be discussed in detail in community meeting.

**Comment by ljkiraly** [comment]
I also tried to eliminate the dst from IP context, but seems to me that this division by zero causes problems: https://github.com/networkservicemesh/sdk-kernel/blob/33c5467b712900bd71fb99c022149383ce04539e/pkg/kernel/tools/heal/liveness_check.go#L50
But I have to double-check.

**Comment by denis-tingaikin** [comment]
@ljkiraly I think we can add something like return true immediately if no addresses in the context

File: main.go [hunk 2]
```diff
@@ -159,16 +160,21 @@ func main() {
 		dnsClient = dnscontext.NewClient(dnscontext.WithChainContext(ctx), dnscontext.WithCorefilePath(c.CoreDNSConfigPath))
 	}
 
+	var healClient = null.NewClient()
+
+	if c.LivenessCheck != "disabled" {
```
Comment (denis-tingaikin) [inline main.go:165]:
```suggestion
	if c.LivenessCheckEnabled {
```

**Review by denis-tingaikin — COMMENTED** [review]
I think that makes sense. Added few comments.

**Review by denis-tingaikin — APPROVED** [review]
Looks good.

@edwarnicke Can we merge this?

File: main.go [hunk 3]
```diff
@@ -159,16 +160,21 @@ func main() {
 		dnsClient = dnscontext.NewClient(dnscontext.WithChainContext(ctx), dnscontext.WithCorefilePath(c.CoreDNSConfigPath))
 	}
 
+	var healClient = null.NewClient()
```
Comment (denis-tingaikin) [inline main.go:163]:
```suggestion
	var healOptions = []heal.Option{ heal.WithLivenessCheckInterval(c.LivenessCheckInterval),}
```

**Review by denis-tingaikin — COMMENTED** [review]
Found a problem

**Comment by ljkiraly** [comment]
@denis-tingaikin Thanks for the suggestions. I updated the branch.
Could you take a look at the remaining PR related to this: networkservicemesh/sdk#1304 ?

File: main.go (hunk 3 above)
Comment (ljkiraly) [reply main.go:163]: I see your point, but what about c.LivenessCheckTimeout? I suppose it should be added here as well:
var healOptions = []heal.Option{ heal.WithLivenessCheckInterval(c.LivenessCheckInterval),
heal.WithLivenessCheckTimeout(c.LivenessCheckTimeout)}
Or?

**Review by denis-tingaikin — APPROVED** [review]

**Pull request merged**